lib/build-asset-map-reader.js

```diff
@@ -24,3 +24,21 @@ function readJSON(assetMap) {
 		});
 	});
 }
+
+// Retries a function that returns a promise
+// The first argument is the function
+// The second argument is an array of ms to wait in between the tries
+// Returns a function that takes the same arguments as the provided function
+function retry(fn, retries) {
+	return (...params) => fn(...params).catch(err => {
+		if(retries.length === 0) {
+			throw err;
+		}
+		let backoff = retries.shift();
+		return wait(backoff).then(_ => retry(fn, retries)(...params));
+	});
+}
```

Comment:
Y U NO TEST? For such slightly mind-bendy stuff, tests tend to provide a bit of confidence.

Reply:
How would a test for that look like with your framework? How can I create the file after a certain number of millis while the process is running?

Reply:
I meant unit tests for the function, i.e. pure JavaScript - which will be (a little) easier once extracted into a separate repo.